File: AdhomPanel/AdhomPanel.Web/Modules/AdhomStore/category/categoryRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace AdhomPanel.AdhomStore
{
    [ConnectionKey("Adham"), Module("AdhomStore"), TableName("[dbo].[Category]")]
    [DisplayName("Category"), InstanceName("Category")]
    [ReadPermission("category")]
    [ModifyPermission("category")]


    [LookupScript]
    public sealed class categoryRow : Row<categoryRow.RowFields>, IIdRow, INameRow
    {
        [DisplayName("Id"), Identity, IdProperty]
        public int? Id
        {
            get => fields.Id[this];
            set => fields.Id[this] = value;
        }

        [DisplayName("Name"), QuickSearch, NameProperty]
        public string Name
        {
            get => fields.Name[this];
            set => fields.Name[this] = value;
        }

        [DisplayName("Description")]
        public string Description
        {
            get => fields.Description[this];
            set => fields.Description[this] = value;
        }

        [DisplayName("Image")]
        public string Image
        {
            get => fields.Image[this];
            set => fields.Image[this] = value;
        }

        public categoryRow()
            : base()
        {
        }

        public categoryRow(RowFields fields)
            : base(fields)
        {
        }

        public class RowFields : RowFieldsBase
        {
            public Int32Field Id;
            public StringField Name;
            public StringField Description;
            public StringField Image;
        }
    }
}

File: AdhomPanel/AdhomPanel.Web/Modules/ServerTypes/AdhomStore/categoryRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface categoryRow {
    Id?: number;
    Name?: string;
    Description?: string;
    Image?: string;
}

export abstract class categoryRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'AdhomStore.category';
    static readonly lookupKey = 'AdhomStore.category';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<categoryRow>('AdhomStore.category') }
    static async getLookupAsync() { return getLookupAsync<categoryRow>('AdhomStore.category') }

    static readonly deletePermission = 'category';
    static readonly insertPermission = 'category';
    static readonly readPermission = 'category';
    static readonly updatePermission = 'category';

    static readonly Fields = fieldsProxy<categoryRow>();
}
